Source/SwiftLintFramework/Rules/LineLengthRule.swift

```diff
@@ -31,15 +31,37 @@ public struct LineLengthRule: ConfigurationProviderRule, SourceKitFreeRule {
     )
 
     public func validate(file: File) -> [StyleViolation] {
-        let minValue = configuration.params.map({ $0.value }).min(by: <)
+        let minValue = configuration.params.map({ $0.value }).min(by: <) ?? Int.max
+        let swiftDeclarationKindsByLine: [[SwiftDeclarationKind]] = file.swiftDeclarationKindsByLine() ?? []
+        let syntaxKindsByLine: [[SyntaxKind]] = file.syntaxKindsByLine() ?? []
         return file.lines.flatMap { line in
             // `line.content.characters.count` <= `line.range.length` is true.
             // So, `check line.range.length` is larger than minimum parameter value.
             // for avoiding using heavy `line.content.characters.count`.
-            if line.range.length < minValue! {
+            if line.range.length < minValue {
                 return nil
             }
 
+            if configuration.ignoresFunctionDeclarations &&
+                line.index < swiftDeclarationKindsByLine.count {
+                let functionKinds = swiftDeclarationKindsByLine[line.index].filter { kind in
+                    SwiftDeclarationKind.functionKinds().contains(kind)
+                }
+                if !functionKinds.isEmpty {
+                    return nil
+                }
+            }
+
+            if configuration.ignoresComments &&
+                line.index < syntaxKindsByLine.count {
```

Comment:
this is almost identical to the implementation of the if before. What do you think about extracting to a private helper function?

Also, we probably should create a `Set` for `SyntaxKind.commentKinds()` and one for `SwiftDeclarationKind.functionKinds()` in private properties so we can get faster lookups (in theory anyway).